wordpress-mobile/gutenberg-mobile
# [NOT READY] Make heading styling distinguishable from paragraph

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/351

This PR updates the style of Heading block based on https://github.com/wordpress-mobile/gutenberg-mobile/issues/351#issuecomment-456522533 

Paragraph and Heading block color changes will be handled in another PR. This PR only improves the font weight and font size for Heading block.

![shadowwithletterspacingiphone7-html](https://user-images.githubusercontent.com/5032900/51901118-fa851300-23c7-11e9-85bd-489330aeb23e.gif)

**The Problem**
We want to use bold font with Heading blocks by default. However, we don't want to generate bold output in the html unless user intentionally selects B option from toolbar. That makes it a bit challenging for us to decide how to make this happen. This PR proposes a solution that mimics bolding behavior by giving extra shadow and letter spacing.

**Implementation Details**
Before this PR Aztec was generating `<strong>` html elements whenever a bold font is detected. We want to override this behavior only for the Heading blocks because heading blocks will be displayed as bold by default even if user doesn't tap B button. So we are overriding this Aztec behavior and  generating `<strong>` html elements if we detect shadow in the text(Shadow is the new bold for heading blocks). 

Aztec PR: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1126
WP iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/10912

**TO TEST**

1 - Testing with Example app:

- Run `yarn install` `yarn start`
- Run `yarn ios` <<< Critical since it'll update Carthage dependencies

**Test 1 - Create a new Heading block**
- Create a new heading block
- As you type verify that the font looks bolder than it was before

**Test 2 - Switch between different H sizes**
- Select part of the text and tap B button
- Verify that the selected area gets bolder
- Switch between H2 H3 H4 types of heading sizes
- Verify that the boldness of selected area is kept

**Test 3 - Verify output html**
- Select part of the text and tap B button
- Switch to HTML mode
- Verfiy only the selected and boldened text is surrounded by `<strong>` tag, and other parts aren't

2 - Testing with WPiOS
PR link: https://github.com/wordpress-mobile/WordPress-iOS/pull/10912
